netbox/extras/conditions.py

```diff
@@ -130,22 +130,30 @@ class ConditionSet:
     :param ruleset: A dictionary mapping a logical operator to a list of conditional rules
     """
     def __init__(self, ruleset):
+
         if type(ruleset) is not dict:
             raise ValueError(f"Ruleset must be a dictionary, not {type(ruleset)}.")
-        if len(ruleset) != 1:
-            raise ValueError(f"Ruleset must have exactly one logical operator (found {len(ruleset)})")
-
-        # Determine the logic type
-        logic = list(ruleset.keys())[0]
-        if type(logic) is not str or logic.lower() not in (AND, OR):
-            raise ValueError(f"Invalid logic type: {logic} (must be '{AND}' or '{OR}')")
-        self.logic = logic.lower()
-
-        # Compile the set of Conditions
-        self.conditions = [
-            ConditionSet(rule) if is_ruleset(rule) else Condition(**rule)
-            for rule in ruleset[self.logic]
-        ]
+
+        self.logic = None
+
+        # If logic type use it, else return the ruleset
+        if len(ruleset) == 1:
```

Comment:
We can use the `is_ruleset()` function, which provides some additional validation, to determine how to treat the root dictionary.

Reply:
I thought to use it, but we could lose the granularity to deal with the errors and the correct error message.